### DISCUSSION TITLE
NetBeans Platform/Java Swing Future

### DISCUSSION THREAD

**Discussion by kenprogrammer** [apache/netbeans#3848] [Q&A] [answered]
I want to switch from developing desktop applications with .NET to Java. But according Oracle Product Roadmap Java **Swing** support may end by **2026**.
Since **NetBeans Platform** relies on this technology, should I go ahead and invest in NetBeans platform/Swing?

**Comment by mbien** [comment, marked as answer]
I don't think swing is going away any time soon, I am not sure where you got the date from, some vendors will probably support Java 8 longer than 2026 (oracle will till at least 2030), not to mention that swing is still in JDK 17 and 19 and is still getting bug fixes and features (https://wiki.openjdk.java.net/display/wakefield/Pure+Wayland+toolkit+plan+proposal).

You can also embed JavaFX in swing, if you want to use newer/different technology. But I don't give career advice on the internet, you have to decide that for yourself.

**Reply by kenprogrammer** [reply]
Thanks for the clarification
